openedx/core/djangoapps/persistenttask/models.py

```diff
@@ -0,0 +1,31 @@
+"""
+Models to support persistent tasks.
+"""
+
+from django.db import models
+
+from openedx.core.lib.properties import JSONProperty
+
+
+class FailedTask(models.Model):
+    """
+    Representation of tasks that have failed
+    """
+    task_name = models.CharField(max_length=255, db_index=True)
+    task_id = models.CharField(max_length=255)
+    argstring = models.TextField(blank=True)
+    kwargstring = models.TextField(blank=True)
+    exc = models.CharField(max_length=255)
+    datetime_failed = models.DateTimeField()
+    datetime_resolved = models.DateTimeField(blank=True, null=True, default=None, db_index=True)
+
+    args = JSONProperty('argstring', allowed_types={list, tuple})
```

Comment:
Looks like there's already a library that does this: https://github.com/bradjasper/django-jsonfield

And it looks like we already use it: requirements/edx/base.txt:jsonfield==1.0.3

Win!